src/transport/PairingSession.h

```diff
@@ -170,18 +177,15 @@ class DLL_EXPORT PairingSession
         mPeerCATs    = kUndefinedCATs;
         mPeerAddress = Transport::PeerAddress::Uninitialized();
         mPeerSessionId.ClearValue();
-        mLocalSessionId = kInvalidKeyId;
+        mSecureSessionHolder.Release();
     }
 
 private:
     const Transport::SecureSession::Type mSecureSessionType;
     NodeId mPeerNodeId = kUndefinedNodeId;
     CATValues mPeerCATs;
 
-    // TODO: the local key id should be allocateed at start
-    // then we can remove kInvalidKeyId
-    static constexpr uint16_t kInvalidKeyId = UINT16_MAX;
-    uint16_t mLocalSessionId                = kInvalidKeyId;
+    SessionHolder mSecureSessionHolder;
```

Comment:
This is just awesome. We now have shifted away from just dangerously holding onto a naked SessionID without a shared ownership model, to one where we now have a legit 'handle' to the session.